File: backend-laravel/app/Http/Controllers/TourismController.php
<?php

namespace App\Http\Controllers;

use App\Services\TouristService;
use Illuminate\Http\JsonResponse;
use Illuminate\Routing\Controller as BaseController;

class TourismController extends BaseController
{
    private TouristService $touristService;
    
    public function __construct(TouristService $touristService)
    {
        $this->touristService = $touristService;
    }
    
    public function getCityInfo(): JsonResponse
    {
        $cityInfo = $this->touristService->getCityInfo();
        return response()->json($cityInfo->toArray());
    }
    
    public function getAllTouristPlaces(): JsonResponse
    {
        $places = $this->touristService->getAllTouristPlaces();
        $placesArray = $places->map(fn($place) => $place->toArray())->toArray();
        return response()->json($placesArray);
    }
    
    public function getTouristPlace(int $id): JsonResponse
    {
        $place = $this->touristService->getTouristPlaceById($id);
        
        if ($place === null) {
            return response()->json([
                'error' => "Tourist place with ID $id not found."
            ], 404);
        }
        
        return response()->json($place->toArray());
    }
    
    public function getTouristPlacesByCategory(string $category): JsonResponse
    {
        $places = $this->touristService->getTouristPlacesByCategory($category);
        $placesArray = $places->map(fn($place) => $place->toArray())->toArray();
        return response()->json($placesArray);
    }
    
    public function getCategories(): JsonResponse
    {
        $categories = $this->touristService->getAllCategories();
        return response()->json($categories);
    }
}

File: backend/Controllers/TourismController.cs
using Microsoft.AspNetCore.Mvc;
using OrtegaTourism.Models;

namespace OrtegaTourism.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class TourismController : ControllerBase
    {
        private readonly TouristService _touristService;

        public TourismController(TouristService touristService)
        {
            _touristService = touristService;
        }

        [HttpGet("city-info")]
        public ActionResult<CityInfo> GetCityInfo()
        {
            return Ok(_touristService.GetCityInfo());
        }

        [HttpGet("places")]
        public ActionResult<List<TouristPlace>> GetAllTouristPlaces()
        {
            return Ok(_touristService.GetAllTouristPlaces());
        }

        [HttpGet("places/{id}")]
        public ActionResult<TouristPlace> GetTouristPlace(int id)
        {
            var place = _touristService.GetTouristPlaceById(id);
            if (place == null)
            {
                return NotFound($"Tourist place with ID {id} not found.");
            }
            return Ok(place);
        }

        [HttpGet("places/category/{category}")]
        public ActionResult<List<TouristPlace>> GetTouristPlacesByCategory(string category)
        {
            var places = _touristService.GetTouristPlacesByCategory(category);
            return Ok(places);
        }

        [HttpGet("categories")]
        public ActionResult<List<string>> GetCategories()
        {
            var categories = _touristService.GetAllTouristPlaces()
                .Select(p => p.Category)
                .Distinct()
                .ToList();
            return Ok(categories);
        }
    }
}